### DISCUSSION TITLE
How to deal with abbreviations?

### DISCUSSION THREAD

**Discussion by johan-horst** [ucum-org/ucum#368] [Q&A] [answered]
Hi,

Currently we are consuming an API that is using the UCUM standard. This algorithm provides for example the unit 'centimeters'. 

Now the question is, how to deal with the abbreviation of centimeters, because we would like to display it in an app, and the word 'centimeter' is to long. I don't see any documentation how to get from 'centimeters' to 'cm'. 
In this case, the code itself is 'cm' but for inches it is '[inch-us]', which is not nice to display. 

So, my question is, is there a standard (documented) way to handle the abbreviations of the units?

Thanks in advance.

**Comment by timbrisc** [comment]
Hello,

May I first ask which API you are using? That should help us identify a solution.

**Reply by johan-horst** [reply]
Hi, 

Thanks for your quick response.

It is a custom made API of an hospital, I can not share you the details of the API itself.
What is can share is a part of an example response: (FHIR QuestionnaireResponse resource)

```
"item": 
[
    {        
        "extension": [
            {
                "url": "http://logicnets.info/fhir/StructureDefinition/questionnaire-isAnswered",
                "valueBoolean": false
            },
            {
                "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-unitOption",
                "valueCoding": {
                    "code": "[in_us]",
                    "display": "inch",
                    "system": "http://unitsofmeasure.org"
                }
            },
            {
                "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-unitOption",
                "valueCoding": {
                    "code": "cm",
                    "display": "centimeter",
                    "system": "http://unitsofmeasure.org"
                }
            }
        ],
        "id": "Q-HeightBMI,
        "linkId": "Q.HeightBMI",
        "required": true,
        "text": "What is your height?",
        "type": "quantity"
    }
],
```

As you can see there are two options given with `http://hl7.org/fhir/StructureDefinition/questionnaire-unitOption`. Both are based on the ucum standard, but as said in my previous message, I would like to have the abbreviation of the unitOption.

Is that somehow supported by the standard?

**Reply by timbrisc** [reply, marked as answer]
Thank you for this additional context. This is not presently supported by UCUM. From §14 of the specification:

> If unit symbols for the purpose of display and print are derived from The Unified Code for Units of Measure units, the square brackets can be removed. However, display units are out of scope of The Unified Code for Units of Measure.

I can foresee display names as a possible normative element of UCUM, however it would be some time before we have an opportunity to adopt this. I have added this as a [feature request](https://github.com/ucum-org/ucum/issues/369) for discussion at a future UCUM Committee meeting.

**Reply by johan-horst** [reply]
Thanks, for now we will implement a custom display.
